packages/liferay-npm-scripts/src/utils/deepMerge.js

```diff
@@ -174,7 +174,22 @@ function babelMerge(key) {
 			);
 		};
 	} else {
-		return combineMerge;
+		return function(target, source, options) {
+			if (Array.isArray(target) && Array.isArray(source)) {
+				return combineMerge(target, source, options);
+			} else if (
+				options.isMergeableObject(target) &&
+				options.isMergeableObject(source)
+			) {
+				return merge(target, source, options);
+			} else {
+				throw new Error(
+					'babelMerge(): ' +
+						'Cannot merge without two mergeable objects: ' +
```

Comment:
One thing I was thinking is that other than stale objects on our part, developers can also mess up their config files, so this should also account for it, right?

In case this also catches that, maybe the message could also point to a syntax error in their config file?

Reply:
Yeah, in theory it will catch some obviously broken config files; for example — not tested, but I think — this will blow up if you have something like this in your config file:

```json
{
    presets: null
}
```

It probably won't catch subtle cases though, and it might be tricky to provide the right guidance other than "Your config file doesn't merge cleanly: please look at it an fix it".

> a syntax error in their config file

If we're literally talking about invalid JSON, I think things already blow up long before we get this far.

Reply:
> "Your config file doesn't merge cleanly: please look at it an fix it"

I think this plus the "it might be stale" bit should be more than enough to put them in the right track, don't you?

Reply:
The thing is, thanks to the other fix (the signal handling), it is *probably* not stale. So I'll update the message to say that it needs human review, but I won't use the word "stale".

Reply:
👍

Reply:
Revised the message text in https://github.com/liferay/liferay-npm-tools/pull/243/commits/985d9a59a71c9fcb8868df72ec8cc2aebeb3542c